src/cppmangle.c

```diff
@@ -164,7 +164,8 @@ class CppMangleVisitor : public Visitor
                         else
                         {
                             dinteger_t val = e->toInteger();
-                            if (val < 0)
+                            // TODO: try and do this without knowing what dinteger_t type is
+                            if ((long)val < 0)
```

Comment:
I mean, this is a bug, right?  Negative C++ template value parameters are currently being mangled incorrectly.  It would be nice to have a test case to go with this fix.